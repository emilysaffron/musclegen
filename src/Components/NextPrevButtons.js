import styled from "@emotion/styled";

const StyledButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  background: #cea8bb;
  width: 125px;
  height: 75px;
  padding: 20px;
  margin: 10px;
  font-size: 20px;
  cursor: pointer;
  outline: none;
`;

const NextPrevButtons = ({
  currentExerciseNumber,
  switchCurrentExerciseNumber,
  maxExerciseNumber,
  direction,
}) => {
  const handleNext = () => {
    if (currentExerciseNumber < maxExerciseNumber - 1) {
      switchCurrentExerciseNumber(currentExerciseNumber + 1);
    }
  };
  const handlePrevious = () => {
    if (currentExerciseNumber > 0) {
      switchCurrentExerciseNumber(currentExerciseNumber - 1);
    }
  };

  if (direction === "next") {
    return <StyledButton onClick={handleNext}>Next</StyledButton>;
  }
  if (direction === "previous") {
    return <StyledButton onClick={handlePrevious}>Previous</StyledButton>;
  }
};

export default NextPrevButtons;
