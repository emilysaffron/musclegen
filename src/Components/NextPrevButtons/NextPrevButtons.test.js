import renderer from "react-test-renderer";
import NextPrevButtons from "./NextPrevButtons";

it("NextPrevButtons Snapshot", () => {
  const tree = renderer
    .create(
      <>
        <NextPrevButtons
          currentExerciseNumber={1}
          switchCurrentExerciseNumber="switchCurrentExerciseNumber"
          maxExerciseNumber={10}
          direction="next"
        />
        <NextPrevButtons
          currentExerciseNumber={1}
          switchCurrentExerciseNumber="switchCurrentExerciseNumber"
          maxExerciseNumber={10}
          direction="previous"
        />
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
