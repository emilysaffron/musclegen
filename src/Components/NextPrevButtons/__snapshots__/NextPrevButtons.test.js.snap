// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextPrevButtons Snapshot 1`] = `
Array [
  <button
    className="css-m31fz8"
    onClick={[Function]}
  >
    Next
  </button>,
  <button
    className="css-m31fz8"
    onClick={[Function]}
  >
    Previous
  </button>,
]
`;
